viewer/src/views/common/cad/GlobalNodeAppearance.ts

```diff
@@ -0,0 +1,14 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+export type GlobalColorDelegate = (
+  modelIdentifier: string,
+  treeIndex: number
+) => [number, number, number, number] | undefined;
+export type GlobalVisibilityDelegate = (modelIdentifier: string, treeIndex: number) => boolean;
+
+export interface GlobalNodeAppearance {
```

Comment:
What is this used for? We should avoid using `Node` about `Models` as we use it about individual 3D nodes (yes, we should rename `CadNode`).

Reply:
I don't think we need a global override method for appearance.

Reply:
Node appearance is logic used for say changing the colors of a node when selected. RevealManager (Base) should have a default picking handler implemented. It doesn't make sense that the user needs to for each model they have define and reuse the same appearance?

Reply:
Then I think that at least they should implement the same interface to avoid the need for two different types of callbacks. It's fine to always pass the model, even if it's just one model. This will simplify the code for applying colors.

However, I think we should send `CadModel` rather than the ID which doesn't have any meaning (?). Any reason not to?

Reply:
Due to the complexity of MaterialManager (it is being used both at the creation stage of an object, pre cadnode creation, and later at transformation from Sector and Quads to THREE.Group). we are creating a map that has a string identifier which links to a set of materials. The map's only functionality is to make sure that we are able to get the materials of a certain cadnode when we are required to modify them. I am not a fan of the implementation, and I do agree that it would be nice to change it, but I currently don't have a good solution for it. Also, the reason why we need two different sets of callbacks is because of how treeindex works. In a multimodel scenario you can't only depend on the tree index because it is the same across multiple models, hence why you need a separate identifier for the model. Say if you create a list next to the viewer and you select a node, you would need to in some way allow that node to change appearance in the 3d viewer, and that logic belongs outside of the manager itself.

Reply:
I understand that in one of the cases you need to identify the model, but why not just provide this to both callbacks?